pkg/client/interface.go

```diff
@@ -98,33 +99,69 @@ type TxContext interface {
 	) (*comettypes.ResultTx, error)
 }
 
-// BlocksObservable is an observable which is notified with an either
-// value which contains either an error or the event message bytes.
-//
-// TODO_HACK: The purpose of this type is to work around gomock's lack of
-// support for generic types. For the same reason, this type cannot be an
-// alias (i.e. EventsBytesObservable = observable.Observable[either.Either[[]byte]]).
-type BlocksObservable observable.ReplayObservable[Block]
-
-// BlockClient is an interface which provides notifications about newly committed
-// blocks as well as direct access to the latest block via some blockchain API.
-type BlockClient interface {
-	// CommittedBlocksSequence returns an observable which emits newly committed blocks.
-	CommittedBlocksSequence(context.Context) BlocksObservable
-	// LatestBlock returns the latest block that has been committed.
-	LatestBlock(context.Context) Block
-	// Close unsubscribes all observers of the committed blocks sequence observable
-	// and closes the events query client.
-	Close()
-}
-
 // Block is an interface which abstracts the details of a block to its minimal
 // necessary components.
 type Block interface {
 	Height() int64
 	Hash() []byte
 }
 
+// Redelegation is an interface which wraps the EventRedelegation event
+// emitted by the application module.
+type Redelegation interface {
+	AppAddress() string
+}
+
+// EventsObservable is a replay observable for events of some type T.
```

Comment:
It's probably worth adding TODO about how this can't be an alias due to the gomock constraint. I wouldn't go into much detail, just enough to articulate why this is not an alias, which seems like it would be more appriate otherwise.